### DISCUSSION TITLE
Why does rate limiting throw HTTP 500 by default instead of  HTTP 429?

### DISCUSSION THREAD

**Discussion by sander1095** [dotnet/aspnetcore#58768] [Q&A] [answered]
Question is based on this context

https://x.com/mjovanovictech/status/1852633606558040199

**Comment by gfoidl** [comment, marked as answer]
By defaults it's HTTP 503 (https://github.com/dotnet/aspnetcore/blob/4ce2db7b8d85c07cad2c59242edc19af6a91b0d7/src/Middleware/RateLimiting/src/RateLimiterOptions.cs#L41)

HTTP 429 is for a specific client, whilst HTTP 503 is more "general". The framework can't know if it's client specific or not (depends on configuration of the rate limiter), thus 503 is used -- and therefore it's also configurable to set HTTP 429 if one knows it's client specific.
